pkg/build/controller/factory/factory.go

```diff
@@ -305,11 +313,20 @@ func (factory *ImageChangeControllerFactory) Create() controller.RunnableControl
 		),
 		Handle: func(obj interface{}) error {
 			imageRepo := obj.(*imageapi.ImageStream)
-			return imageChangeController.HandleImageRepo(imageRepo)
+			return imageChangeController.HandleImageStream(imageRepo)
 		},
 	}
 }
 
+func (factory *ImageChangeControllerFactory) waitForSyncedStores() {
+	for !factory.BuildConfigIndexSynced() {
```

Comment:
I would suggest rewritting the controller to use Informers instead of the RunnableController framework. Currently nothing initializes the cache so it never syncs. By using the shared informer framework, the master will do it for you.